modules/Blockchain/XDai/index.js

```diff
@@ -0,0 +1,1558 @@
+const fs = require('fs');
+const BN = require('bn.js');
+const uuidv4 = require('uuid/v4');
+const ethereumAbi = require('ethereumjs-abi');
+const Op = require('sequelize/lib/operators');
+const Web3 = require('web3');
+
+const Transactions = require('./Transactions');
+const Utilities = require('../../Utilities');
+const Models = require('../../../models');
+const path = require('path');
+const constants = require('../../constants');
+
+class Ethereum {
+    /**
+     * Initializing Ethereum blockchain connector
+     */
+    constructor({
+        config, emitter, logger, gasStationService, tracPriceService,
+    }, configuration) {
+        this.contractsLoaded = false;
+        this.initialized = false;
+
+
+        if (process.env.RPC_SERVER_URL) {
+            configuration.rpc_server_url = process.env.RPC_SERVER_URL;
+        }
+
+        if (!configuration.rpc_server_url) {
+            console.error('Please provide a valid RPC server URL.\n' +
+                'Add it to the blockchain section. For example:\n' +
+                '   "blockchain": {\n' +
+                '       "rpc_server_url": "http://your.server.url/"\n' +
+                '   }');
+            return;
+        }
+
+        // Loading Web3
+        this.emitter = emitter;
+        this.web3 = new Web3(new Web3.providers.HttpProvider(configuration.rpc_server_url));
+        this.logger = logger;
+        this.gasStationService = gasStationService;
+        this.tracPriceService = tracPriceService;
+
+        this.config = configuration;
+        this.config.appDataPath = config.appDataPath;
+        const walletObject = Utilities.loadJsonFromFile(
+            this.config.appDataPath,
+            configuration.node_wallet_path,
+        );
+        if (walletObject) {
+            const {
+                node_wallet,
+                node_private_key,
+                management_wallet,
+            } = walletObject;
+
+            this.config.wallet_address = node_wallet;
```

Comment:
We are reading wallets from config, not from separate file. Also we should remove usage of this.config.wallet_address